pageserver/src/remote_storage/storage_sync/download.rs

```diff
@@ -43,53 +43,67 @@ pub(super) async fn download_timeline<
     sync_id: TimelineSyncId,
     mut download: TimelineDownload,
     retries: u32,
-) -> Option<bool> {
+) -> Option<(bool, Lsn)> {
```

Comment:
It would be good to document what is the meaning of different return values. I e None vs Some(false)